sdk/python/kfp/components/pipeline_task.py

```diff
@@ -246,6 +246,13 @@ def set_caching_options(self, enable_caching: bool) -> 'PipelineTask':
         self._task_spec.enable_caching = enable_caching
         return self
 
+    def _ensure_container_spec_exists(self) -> None:
```

Comment:
Micro improvement suggestion: Could also use a `method_name: str` param to which you pass `self.set_memory_limit.__name__` (or whatever the method is) to include the method name in the error message.

Reply:
Can't really reference `self.func` from within the same `func`. Opted to use inspect to achieve the same goal.

Reply:
SG. `inspect` works!

Here's what I had in mind. Curious, where did things break down?


```python
class Cls:
    def method(self):
        self.printer(self.method.__name__)

    def printer(self, name):
        print(name)

Cls().method()
>>> 'method'
```

Reply:
Okay, that works, I thought I tried the same and got the error but maybe I missed something else.
Anyway, I think I'll keep the inspect approach in favor of there's less effort for the caller to pass in values.